### DISCUSSION TITLE
Debug doesn't hypen UUID's anymore

### DISCUSSION THREAD

**Discussion by Jasperav** [uuid-rs/uuid#601] [Q&A]
The Display will still use hyphens in the UUID, but the Debug trait doesn't. I am logging a lot of information, and since I think version 1.0, I don't have the hyphens in the logging anymore. Now I don't know how I can convert the unhyphened uuid to a 'normal' uuid so I can debug easily.

Why is doesn't the Debug trait include hyphens anymore? Is there an easy way to get it back? I don't want to rewrite my entire codebase to use the Hyphenated struct.

**Comment by KodrAus** [comment]
Hi @Jasperav :wave:

The `Debug` impl omits hyphens when formatted using the alternate format `{:#?}`, but still includes hyphens when using the standard debug format `{:?}`. Are you trying to print UUIDs included in bigger data structures that you're formatting using `{:#?}` to get more whitespace?

**Reply by Jasperav** [reply]
Thanks for the response @KodrAus. I don't understand why the hyphens are removed when using the pretty debug print, and are present using the 'regular' debug print. The pretty debug print looks nicer for me, I use it for projects. Debugging is a lot easier when having the hyphens in place when using the pretty debug print. Is there any specific reason why pretty printing removes the hyphens?

**Reply by KodrAus** [reply]
I think the problem here is that the flag is simply exposed as "alternate" formatting, rather than specifically for pretty-printing. But the most common use of it would be to pretty-print debug structs, which I think is a good enough reason to revert that change. The `Debug` format isn't considered stable so we can do that.

If you'd like to submit a PR to fix that up please feel free!
